analyses/rna-seq-expression-summary-stats/README.md

```diff
@@ -0,0 +1,181 @@
+
+### Purpose
+
+Within each cancer group and cohort, calculate TPM means, standard deviations, z-scores, and ranks.
+
+### Methods
+
+Select independent RNA-seq samples using `independent-specimens.rnaseq.primary.tsv` in the results of the `independent-samples` analysis module.
+
+Combine CBTN and PNOC into one cohort, PBTA.
+
+For each `cancer_group`, select one of the following two sets of samples:
+
+- Samples from all `cohort`s, e.g. PBTA and GMKF.
+- Samples from each individual `cohort`.
+
+If >= 5 samples are selected, generate the following summary statistics:
+
+- TPM means of each gene across all selected samples, and denote this vector as `mean_TPM_vector`.
+- TPM standard deviations of each gene across all selected samples.
+- z-scores of each gene across all genes, computed as `z_score_vector = (mean_TPM_vector - mean(mean_TPM_vector)) / sd(mean_TPM_vector)`. Call these z-scores as `mean_tpm_cancer_group_wise_z_scores` in the filenames.
+- Ranks of mean TPM of each gene across all selected samples, which takes one of the following four values: `Highest expressed 25%`, `Expression between upper quartile and median`, `Expression between median and lower quartile`, and `Lowest expressed 25%`. If multiple genes have the same mean TPM value, their tied rank is the lowest rank, in order to be conservative on the description of their expression levels.
+
+Combine each type of the summary statistics vectors into a table, with rows as genes, and columns as `cancer_group_cohort`.
+
+Denote the (`n_genes`, `n_cancer_groups`/`n_cancer_group_cohorts`) `mean_TPM_vector` combined matrix as `mean_TPM_matrix`.
+
+Generate z-scores across all `cancer_groups`/`cancer_group_cohorts` as `z_score_matrix = (mean_TPM_matrix - rowMeans(mean_TPM_matrix)) / rowSD(mean_TPM_matrix)`. Call these z-scores as `mean_tpm_gene_wise_z_scores` in the filenames.
+
+Call aforementioned (`n_genes`, `n_cancer_groups`/`n_cancer_group_cohorts`) tables as wide tables.
+
+Generate long summary statistic tables for converting to JSON format, with each row as a tab-delimited record of the following columns, as suggested by @jharenza and @taylordm at <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/27#issuecomment-868035006>.
+
+- `gene_symbol`
+- `gene_id`
+- `cancer_group`
+- `cohort`
+- `tpm_mean`
+- `tpm_sd`
+- `tpm_mean_cancer_group_wise_zscore`/`tpm_mean_gene_wise_zscore`
+- `tpm_mean_cancer_group_wise_quantiles`
+- `n_samples`
+
+### Results
+
+#### All cohort summary statistics tables
+
+The following wide tables are generated using the methods described above. Rows are genes. Columns are `cancer_group`s, except that the first two columns are gene symbol and gene Ensembl ID. If one gene symbol matches to multiple Ensembl IDs, the value of the Ensembl ID column is a comma separated list of all Ensembl IDs, e.g. `ENSG00000206952.3,ENSG00000281910.1`. In `inpiut/ens_symbol.tsv`, `SNORA50A` is mapped to both `ENSG00000206952.3` and `ENSG00000281910.1`.
```

Comment:
I'm sure you will be updating this

Reply:
Thanks for the note. Will update in the next patch.